hal/src/rtl872x/spi_hal.cpp

```diff
@@ -445,7 +470,6 @@ class Spi {
 
     uint8_t transfer(uint8_t data) {
         CHECK_TRUE(isEnabled(), 0);
-        CHECK_TRUE(isBusy() == false, 0);
         CHECK_TRUE(config_.spiMode == SPI_MODE_MASTER, 0);
```

Comment:
I know this isnt a change you did in this PR, but shouldnt we return a non-zero error code if we are *not* in master mode? IE something like this:
```
CHECK_TRUE(config_.spiMode == SPI_MODE_MASTER, SYSTEM_ERROR_INVALID_STATE);
```
Or am I misunderstanding `CHECK_TRUE()`?

Reply:
It is consistent with Gen3. And the wiring API `SPI.transfer()` returns an unsigned `byte` value, so it's fine to return `0` here.

Reply:
Ok, makes sense. Returning `0` to indicate no data was sent seems correct. I was misinterpreting it as `SYSTEM_ERROR_NONE`